src/main/java/teammates/logic/core/StudentsLogic.java

```diff
@@ -628,22 +620,32 @@ private StudentEnrollDetails enrollStudent(StudentAttributes validStudentAttribu
         return studentList;
     }
 
-    private List<String> getInvalidityInfoInDuplicatedEmail(String email,
-            List<String> studentEmailList, String[] linesArray) {
-        List<String> info = new ArrayList<>();
-        info.add("Same email address as the student in line \"" + linesArray[studentEmailList.indexOf(email) + 1] + "\"");
-        return info;
+    private String getInvalidityInfoInDuplicatedEmail(String[] linesArray, int duplicateEmailIndex) {
+        return "Same email address as the student in line \"" + linesArray[duplicateEmailIndex + 1] + "\"";
     }
 
-    private boolean isStudentEmailDuplicated(String email,
-            List<String> studentEmailList) {
-        return studentEmailList.contains(email);
+    private String addInvalidStudentInfo(StudentAttributes student, String sanitizedLine) {
```

Comment:
Also, update the method name. The caller is doing `invalidityInfo.add(addInvalidStudentInfo(...));`, which doesn't read off well. Maybe something like `invalidStudentInfo(...)` will be better.